muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/pedagogy/rest/PedagogyRestService.java

```diff
@@ -0,0 +1,421 @@
+    PedagogyForm form = pedagogyController.findFormByStudentIdentifier(studentIdentifier);
+    return Response.ok(form == null ? PedagogyFormState.INACTIVE : form.getState()).build();
+  }
+
+  /**
+   * mApi().pedagogy.form.state.update('PYRAMUS-STUDENT-123', {state: String});
+   */
+  @Path("/form/{STUDENTIDENTIFIER}/state")
+  @PUT
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateState(@PathParam("STUDENTIDENTIFIER") String studentIdentifier, PedagogyFormStatePayload payload) {
+    
+    // Payload validation
+    
+    PedagogyForm form = pedagogyController.findFormByStudentIdentifier(studentIdentifier);
+    if (form == null) {
+      return Response.status(Status.NOT_FOUND).entity(String.format("Form for student %s not found", studentIdentifier)).build();
+    }
+    
+    // Access check
+    
+    PedagogyFormAccessRestModel access = getAccess(studentIdentifier, true);
+    if (!access.isAccessible()) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+
+    // Valid change; only supports ACTIVE -> PENDING by staff and PENDING -> APPROVED by form student
+    
+    boolean validChange = false;
+    if (payload.getState() == PedagogyFormState.PENDING) {
+      validChange = form.getState() == PedagogyFormState.ACTIVE; 
+    }
+    else if (payload.getState() == PedagogyFormState.APPROVED) {
+      validChange = form.getState() == PedagogyFormState.PENDING && StringUtils.equals(sessionController.getLoggedUser().toId(), form.getStudentIdentifier()); 
+    }
+    if (!validChange) {
+      return Response.status(Status.BAD_REQUEST).entity("Invalid state change").build();
+    }
+    
+    // State update
+    
+    form = pedagogyController.updateState(form, payload.getState(), sessionController.getLoggedUserEntity().getId());
+    
+    return Response.ok(toRestModel(form)).build();
+  }
+  
+  private PedagogyFormRestModel toRestModel(PedagogyForm form) {
+    return toRestModel(form, form.getStudentIdentifier());
+  }
+
+  private PedagogyFormRestModel toRestModel(PedagogyForm form, String studentIdentifier) {
+    PedagogyFormRestModel model = new PedagogyFormRestModel();
+    
+    // Owner and student contact info
+    
+    schoolDataBridgeSessionController.startSystemSession();
+    try {
+      SchoolDataIdentifier identifier = SchoolDataIdentifier.fromId(studentIdentifier);
+      UserContactInfo contactInfo = userController.getUserContactInfo(identifier.getDataSource(), identifier.getIdentifier());
+      model.setStudentInfo(toMap(contactInfo));
+      Long formCreator = getFormCreator(form);
+      if (formCreator != null) {
+        model.setOwnerId(formCreator);
+        UserEntity userEntity = userEntityController.findUserEntityById(formCreator);
+        if (userEntity != null) {
+          contactInfo = userController.getUserContactInfo(userEntity.getDefaultSchoolDataSource().getIdentifier(), userEntity.getDefaultIdentifier());
```

Comment:
Does the form really need all this information about the creator/owner (birthday, address, phone email etc)? Especially if the creator/owner can be a staff member.